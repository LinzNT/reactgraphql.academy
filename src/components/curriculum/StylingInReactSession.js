import React from 'react'
import Ul, { Li } from '../layout/Ul'
import Session from './Session'

const StylingInReactSession = ({ hideTitle }) => (
  <Session hideTitle={hideTitle} title="Styling in React">
    <Ul>
      <Li>CSS Vs. JS</Li>
      <Li>Styled-components</Li>
      <Li>
        Component libraries comparison: SemanticUI, MaterialUI, Rebass, and
        React-Bootstrap
      </Li>
    </Ul>
  </Session>
)

export default StylingInReactSession
