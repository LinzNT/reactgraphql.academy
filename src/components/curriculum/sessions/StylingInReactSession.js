import React from 'react'
import Ul, { Li } from '../../layout/Ul'
import Session from './Session'

const StylingInReactSession = ({ title = 'Styling in React' }) => (
  <Session title={title}>
    <Ul>
      <Li>
        CSS approaches:
        <Ul>
          <Li>CSS at run time vs CSS at build time</Li>
          <Li>Inline style vs CSS-in-JS</Li>
        </Ul>
      </Li>
      <Li>
        UI component libraries comparison: SemanticUI, MaterialUI, Rebass, and
        React-Bootstrap
      </Li>
      <Li>Styled-components</Li>
      <Li>Storybook</Li>
    </Ul>
  </Session>
)

export default StylingInReactSession
