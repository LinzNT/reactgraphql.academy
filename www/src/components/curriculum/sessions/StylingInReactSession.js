import React from 'react'
import Ul, { Li } from '../../layout/Ul'
import Session from './Session'

export const titleSession = 'Component library fundamentals'

const StylingInReactSession = ({ title = titleSession }) => (
  <Session title={title}>
    <Ul>
      <Li>CSS-in-JS with Styled-components</Li>
      <Li>
        UI component libraries comparison: SemanticUI, MaterialUI, Rebass, and
        React-Bootstrap
      </Li>
      <Li>Storybook</Li>
      <Li>
        Context
        <Ul>
          <Li>Modals</Li>
          <Li>Themes</Li>
        </Ul>
      </Li>
    </Ul>
  </Session>
)

export default StylingInReactSession
