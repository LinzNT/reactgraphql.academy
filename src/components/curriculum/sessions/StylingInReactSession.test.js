import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import StylingInReactSession from './StylingInReactSession'

describe('<StylingInReactSession />', () => {
  it('should render component', () => {
    const wrapper = shallow(<StylingInReactSession />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
