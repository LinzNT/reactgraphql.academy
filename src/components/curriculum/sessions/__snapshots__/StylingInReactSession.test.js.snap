// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StylingInReactSession /> should render component 1`] = `
<Session>
  <styled.ul>
    <styled.li>
      CSS Vs. JS
    </styled.li>
    <styled.li>
      Styled-components
    </styled.li>
    <styled.li>
      Component libraries comparison: SemanticUI, MaterialUI, Rebass, and React-Bootstrap
    </styled.li>
    <styled.li>
      Storybook
    </styled.li>
  </styled.ul>
</Session>
`;
